lstchain/datachecks/containers.py

```diff
@@ -173,35 +173,38 @@ def fill_event_wise_info(self, subrun_index, table, mask, geom,
                      bins=histogram_binnings.hist_dist0_intensity_gt_200)
         self.hist_dist0_intensity_gt_200 = counts
 
-        counts, _, _, _ = plt.hist2d(table['intensity'][mask],
-                                     table['width'][mask],
+        counts, _, _, _ = plt.hist2d(intensity, np.array(table['width'][mask]),
                                      bins=histogram_binnings.hist_width)
         self.hist_width = counts
 
-        counts, _, _, _ = plt.hist2d(table['intensity'][mask],
-                                     table['length'][mask],
+        counts, _, _, _ = plt.hist2d(intensity, np.array(table['length'][mask]),
                                      bins=histogram_binnings.hist_length)
         self.hist_length = counts
 
-        counts, _, _, _ = plt.hist2d(table['intensity'][mask],
-                                     table['skewness'][mask],
+        counts, _, _, _ = plt.hist2d(intensity,
+                                     np.array(table['skewness'][mask]),
                                      bins=histogram_binnings.hist_skewness)
         self.hist_skewness = counts
 
-        intercept = table['intercept'][mask]
         counts, _, _, _ = \
-            plt.hist2d(intensity, intercept,
+            plt.hist2d(intensity, np.array(table['intercept'][mask]),
                        bins=histogram_binnings.hist_intercept)
         self.hist_intercept = counts
 
         length = table['length'][mask]
         tgrad = np.abs(table['time_gradient'][mask])
         counts, _, _, _ = \
-            plt.hist2d(length, tgrad,
+            plt.hist2d(np.array(length), np.array(tgrad),
                        bins=histogram_binnings.hist_tgrad_vs_length)
         self.hist_tgrad_vs_length = counts
+
+        # We noticed an occasional pyplot error that seems to be fixed by
+        # making sure that the coordinates passed to hist2d are ndarrays
+        # (instead of Pandas data series)
+
         counts, _, _, _ = \
-            plt.hist2d(length[intensity > 200], tgrad[intensity > 200],
+            plt.hist2d(np.array(length[intensity > 200]),
```

Comment:
also here, `.to_numpy()` instead of np.array

Reply:
Ok, done!